diff --git a/libavfilter/vf_subtitles.c b/libavfilter/vf_subtitles.c
index 5e1324c..fa15442 100644
--- a/libavfilter/vf_subtitles.c
+++ b/libavfilter/vf_subtitles.c
@@ -371,7 +371,7 @@ static av_cold int init_subtitles(AVFilterContext *ctx)
     }
 
     /* Initialize fonts */
-    ass_set_fonts(ass->renderer, NULL, NULL, 1, NULL, 1);
+    ass_set_fonts(ass->renderer, "default.ttf", NULL, 1, NULL, 1);
 
     /* Open decoder */
     dec_ctx = st->codec;
