diff --git a/libavutil/time.c b/libavutil/time.c
index dbaee02..38a17cc 100644
--- a/libavutil/time.c
+++ b/libavutil/time.c
@@ -55,7 +55,7 @@ int64_t av_gettime(void)
 
 int64_t av_gettime_relative(void)
 {
-#if HAVE_CLOCK_GETTIME && defined(CLOCK_MONOTONIC)
+#if 0 && HAVE_CLOCK_GETTIME && defined(CLOCK_MONOTONIC)
     struct timespec ts;
     clock_gettime(CLOCK_MONOTONIC, &ts);
     return (int64_t)ts.tv_sec * 1000000 + ts.tv_nsec / 1000;
@@ -66,7 +66,7 @@ int64_t av_gettime_relative(void)
 
 int av_gettime_relative_is_monotonic(void)
 {
-#if HAVE_CLOCK_GETTIME && defined(CLOCK_MONOTONIC)
+#if 0 && HAVE_CLOCK_GETTIME && defined(CLOCK_MONOTONIC)
     return 1;
 #else
     return 0;
