diff --git a/libavformat/file.c b/libavformat/file.c
index 08c7f8e6dd..3ce026d303 100644
--- a/libavformat/file.c
+++ b/libavformat/file.c
@@ -38,6 +38,8 @@
 #include "os_support.h"
 #include "url.h"
 
+extern int emscripten_stdin_async(unsigned char* buf, int size);
+
 /* Some systems may not have S_ISFIFO */
 #ifndef S_ISFIFO
 #  ifdef S_IFIFO
@@ -111,7 +113,11 @@ static int file_read(URLContext *h, unsigned char *buf, int size)
     FileContext *c = h->priv_data;
     int ret;
     size = FFMIN(size, c->blocksize);
-    ret = read(c->fd, buf, size);
+    if (c->fd == 0) {
+        ret = emscripten_stdin_async(buf, size);
+    } else {
+        ret = read(c->fd, buf, size);
+    }
     if (ret == 0 && c->follow)
         return AVERROR(EAGAIN);
     if (ret == 0)
