diff --git a/build/make/Makefile b/build/make/Makefile
index 3e8c02490..362a37310 100644
--- a/build/make/Makefile
+++ b/build/make/Makefile
@@ -295,8 +295,8 @@ define so_template
 $(1):
 	$(if $(quiet),@echo "    [LD] $$@")
 	$(qexec)$$(LD) -shared $$(LDFLAGS) \
-            -Wl,--no-undefined -Wl,-soname,$$(SONAME) \
-            -Wl,--version-script,$$(EXPORTS_FILE) -o $$@ \
+            -Wl,-soname,$$(SONAME) \
+            -o $$@ \
             $$(filter %.o,$$^) $$(extralibs)
 endef
 
